routes/videos/VideosController.cs

```diff
@@ -263,25 +273,70 @@ public async Task<IActionResult> AddVideoComment(YoutubeContext db, long videoId
             videoId = newComment.Videoid
         });
     }
-    
+
     [HttpGet("{videoId}/details")]
     [AllowAnonymous]
     public async Task<IActionResult> GetVideoDetails(YoutubeContext db, int videoId)
     {
         try
         {
+            var watcherId = User.FindFirst(ClaimTypes.NameIdentifier);
+
             var v = await db.Videos.Include(v => v.Likes).Include(v => v.Owneraccount).SingleAsync(v => v.Id == videoId);
-            var likes = v.Likes.Count;
-            
-            return Ok(new
+            var likes = v.Likes.ToList();
+            var likeCount = likes.FindAll(l => l.Unlike == false).Count();
+            var dislikeCount = likes.FindAll(l => l.Unlike == true).Count();
+
+            if (watcherId == null)
+            {
+                return Ok(new
+                {
+                    id = v.Id,
+                    createdAt = v.CreatedAt,
+                    userId = v.Owneraccountid,
+                    userFullName = v.Owneraccount.Fullname,
+
+                    views = v.Views,
+                    likes = likeCount,
+                    dislikes = dislikeCount,
+                    subscriptions = v.Owneraccount.Subscriptions.Count(),
+
+                    title = v.Title,
+                    description = v.Description,
+                    category = v.Category
+                });
+            }
+            else
             {
-                id = v.Id,
-                createdAt = v.CreatedAt,
-                userId = v.Owneraccountid,
-                
-                views = v.Views,
-                likes = likes
-            });
+                var like = await db.Likes.Where(l => l.Video == v.Id).Where(l => l.Account == long.Parse(watcherId.Value)).ToArrayAsync();
+                var isLiked = false;
+                var isDisliked = false;
+
+                if (like.Length == 1)
+                {
+                    isLiked = !like[0].Unlike;
+                    isDisliked = like[0].Unlike;
+                }
+                return Ok(new
+                {
+                    id = v.Id,
+                    createdAt = v.CreatedAt,
+                    userId = v.Owneraccountid,
+                    userFullName = v.Owneraccount.Fullname,
+
+                    views = v.Views,
+                    likes = likeCount,
+                    dislikes = dislikeCount,
+                    subscriptions = v.Owneraccount.Subscriptions.Count(),
+
+                    title = v.Title,
+                    description = v.Description,
+                    category = v.Category,
+
+                    isLiked = isLiked,
+                    isDisliked = isDisliked
```

Comment:
mozna napisac po prostu
isLiked,
isDisliked

jezeli nazwa parametru w obiekcie jest taka sama jak zmienna

Reply:
ale to w sumie moze tak zostac